components/config/pref_util.rs

```diff
@@ -158,9 +158,12 @@ impl From<PrefValue> for [f64; 4] {
     fn from(other: PrefValue) -> [f64; 4] {
         match other {
             PrefValue::Array(values) if values.len() == 4 => {
-                let mut f = values.into_iter().map(|v| v.try_into());
-                if f.all(|v| v.is_ok()) {
-                    let f = f.flatten().collect::<Vec<f64>>();
+                let f = values
+                    .into_iter()
+                    .map(|v| v.try_into())
+                    .flatten()
+                    .collect::<Vec<f64>>();
```

Comment:
Could this be slightly shorter with:

```suggestion
                    .filter_map(|value| value.try_into())
                    .collect::<Vec<f64>>();
```